editor/modes/text-editor/index.js

```diff
@@ -13,19 +13,21 @@ function TextEditor( { blocks, onChange } ) {
 	return (
 		<div>
 			<header className="editor-text-editor__formatting">
-				<button className="editor-text-editor__bold">b</button>
-				<button className="editor-text-editor__italic">i</button>
-				<button className="editor-text-editor__link">link</button>
-				<button>b-quote</button>
-				<button className="editor-text-editor__del">del</button>
-				<button>ins</button>
-				<button>img</button>
-				<button>ul</button>
-				<button>ol</button>
-				<button>li</button>
-				<button>code</button>
-				<button>more</button>
-				<button>close tags</button>
+				<div className="editor-text-editor__formatting-inner">
```

Comment:
> Move the entire editor canvas into a fixed position container and assign the headers as top-assigned absolute within. It's probably not great for maintainability to have non-root components consider sidebar and admin bar offsets.

This is interesting. I'm a little nervous about using too much fixed position, though, since css transitions (as I've learned and I'm sure you recall because you graciously fixed it for me) affect fixed position. Might there be sideeffects like that here, should we use CSS transitions on content inside the canvas?

Also, in the future we're likely to have more fixed or sticky position stuff like this. For example if you have a very very long text block where you need to scroll, as you scroll down the page, the Quick Toolbar (<a href="https://wpcoredesign.mystagingwebsite.com/gutenberg/#blueprints">new name for the controls docked to the top of the block</a>) would unstick and dock themselves to the top of the screen instead. Like the Calypso editor, in a way. 

I'm not objecting, just trying to wrap my head around all the components. 

> Use a different unit for the padding: a pixel or vw value perhaps

The problem isn't so much the "faux-centering" the padding does of the contents within, it's rather that on very large breakpoints I mean to apply a max-width to the text inside the editor:

![screen shot 2017-04-14 at 10 15 10](https://cloud.githubusercontent.com/assets/1204802/25037788/58d23c70-20fb-11e7-857e-5e68992d490b.png)

I should have clarified this in the past, sorry. But this is the code in question:

```
.editor-text-editor__formatting,
.editor-text-editor {
	padding-left: $item-spacing * 2;
	padding-right: $item-spacing * 2;
	max-width: $break-huge;
	margin: 0 auto;

	@include break-large() {
		padding-left: 10%;
		padding-right: 10%;
	}

	@include break-wide() {
		padding-left: 15%;
		padding-right: 15%;
	}

	@include break-huge() {
		padding-left: 200px;
		padding-right: 200px;
	}
}
```

As you can see in the `break-huge` breakpoint (1440px), the padding becomes a px unit. Combined with a max-width also of 1440 and `margin: 0 auto;`, that centers the text editor and doesn't let it grow larger. This is working fine, but as you can see on the docked toolbar, the gray border below it doesn't extend fully left and right. I suppose I could potentially do some `:before` magic to do that, instead of having a superfluous container. But that, specifically, is the problem I'm trying to solve.